src/components/Editor/HighlightLine.js

```diff
@@ -59,17 +86,22 @@ export class HighlightLine extends PureComponent<Props> {
     selectedFrame: Frame,
     selectedSource: SourceRecord
   ) {
-    if (!isDocumentReady(selectedSource, selectedLocation)) {
+    const { sourceId, line } = selectedLocation;
+
+    const editorLine = toEditorLine(sourceId, line);
+    if (this.isInSteppingUpdate && editorLine === this.previousEditorLine) {
       return;
     }
+    this.isInSteppingUpdate = false;
+    this.previousEditorLine = editorLine;
 
-    const { sourceId, line } = selectedLocation;
-
+    if (!isDocumentReady(selectedSource, selectedLocation)) {
```

Comment:
could this condition move?

Reply:
I added it to `shouldComponentUpdate`, not sure if I should remove it from here as well since React can render despite the results of `shouldComponentUpdate` from what I understand.

Reply:
From what I have tried, `if (!line || isDebugLine(selectedFrame, selectedLocation))` cannot be put in `shouldComponentUpdate` since it needs to run the remove the current highlight.